src/Microsoft.TemplateEngine.Edge/Settings/GlobalSettings.cs

```diff
@@ -1,42 +1,198 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#nullable enable
+
 using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.TemplateEngine.Abstractions;
 using Microsoft.TemplateEngine.Abstractions.GlobalSettings;
 using Newtonsoft.Json;
 
 namespace Microsoft.TemplateEngine.Edge.Settings
 {
-    public class GlobalSettings : IGlobalSettings
+    class GlobalSettings : IGlobalSettings, IDisposable
```

Comment:
consider the proper Dispose pattern for non sealed classes
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

Reply:
sealed it :)